svm/api.go

```diff
@@ -134,3 +138,67 @@ func (svm *SVM) ValidateAndAddTxToPool(tx *types.Transaction) error {
 	}
 	return nil
 }
+
+// HandleGossipTransaction wraps around HandleTransaction,
+// which handles data received on the transactions gossip channel.
+func (svm *SVM) HandleGossipTransaction(ctx context.Context, _ p2p.Peer, msg []byte) pubsub.ValidationResult {
+	tx, err := types.BytesToTransaction(msg)
+	if err != nil {
+		svm.state.With().Error("SVM couldn't parse incoming transaction", log.Err(err))
+		return pubsub.ValidationIgnore
+	}
+
+	if err = svm.HandleTransaction(tx, true); err != nil {
+		svm.state.With().Error("handle transaction", log.Err(err))
+		return pubsub.ValidationIgnore
+	}
+
+	return pubsub.ValidationAccept
+}
+
+// HandleTxSyncData handles data received on transaction sync.
+func (svm *SVM) HandleTxSyncData(data []byte) error {
+	var tx mesh.DbTransaction
+	err := types.BytesToInterface(data, &tx)
+	if err != nil {
+		svm.state.With().Error("SVM couldn't parse incoming transaction", log.Err(err))
+		return fmt.Errorf("parse: %w", err)
+	}
+	if err = svm.HandleTransaction(tx.Transaction, false); err != nil {
+		return fmt.Errorf("handle transaction: %w", err)
+	}
+	return nil
+}
+
+// HandleTransaction handles data received on transaction gossip channel.
+// It allows for transactions from both valid and invalid blocks to be added to the mempool.
+func (svm *SVM) HandleTransaction(tx *types.Transaction, validateOrigin bool) error {
```

Comment:
I don't think we should export this method (in Go, functions that start with capital letters are made available outside the package). This should be private.

Reply:
Understandable. I exported it because the function signature (as described in the underlying issue) appears so.